### PR TITLE
Update Deltacode documentation as part of Skeleton upgrade

### LINKED ISSUE #188: Update documentation after deltacode gets merge in scancode-toolkit

### PR DESCRIPTION
**Pull request by arnav-mandal1234**
* Modify `docs/source/deltacode/comprehensive_installation.rst`.
 * Modify `docs/source/deltacode/development.rst`.
 * Modify `docs/source/index.rst`.
 * Modify `docs/source/deltacode/GSOC_2021_report.rst`.
 * Modify `README.rst`
 * Fixes https://github.com/nexB/deltacode/issues/188

### CODE REVIEW

File: docs/source/deltacode/comprehensive_installation.rst [hunk 1]
```diff
@@ -1,26 +1,56 @@
...
+
+- DeltaCode requires a Python version 3.8, 3.9 or 3.10 and is
+  tested on Linux, macOS, and Windows. It should work fine on FreeBSD.
 
 System Requirements
 -------------------
 
-- Hardware : DeltaCode will run best with a modern X86 processor and at least 1GB of RAM and
-  250MB of disk.
+- Hardware : DeltaCode will run best with a modern X86 64 bits processor and at
```
Comment (arnav-mandal1234) [inline docs/source/deltacode/comprehensive_installation.rst:35]: Copied this from scancode-toolkit.

File: README.rst [hunk 2]
```diff
@@ -28,39 +19,61 @@ locations.
...
+  <https://deltacode.readthedocs.io/en/latest/comprehensive_installation.html#source-code-install>`_
+
+- `Development installation as a library with "pip install deltacode"
+  <https://deltacode.readthedocs.io/en/latest/comprehensive_installation.html#pip-install>`_
+
+- `Run in a Docker container with a git clone and "docker run"
+  <https://deltacode.readthedocs.io/en/latest/comprehensive_installation.html#docker-install>`_
+
+
+Quick Start
```
Comment (JonoYang) [inline README.rst:62]: I think there should be a sentence mentioning to run `./configure` or `./configure --dev` prior to running `deltacode`

**Review by JonoYang — CHANGES_REQUESTED** [review]
Looks good, I left a suggestion.

File: README.rst (hunk 2 above)
Comment (arnav-mandal1234) [reply README.rst:62]: The above section of `Quick Start` is `Installation`. It has all the installation guides and I have linked it to the documentation page. See: `Development installation from source code using a git clone`. Moreover, `Quick Start` section is to guide everyone on how to use Deltacode.

I have kept the structure same as scancode-toolkit. Let me know what do you think?

Comment (JonoYang) [reply README.rst:62]: Ah, I see what you mean. That makes sense.

**Review by JonoYang — APPROVED** [review]
LGTM! Thanks for updating the doc!

**Comment by AyanSinhaMahapatra** [comment]
LGTM!
Thanks ++ @arnav-mandal1234 , merging!

**Pull request merged**